Pages/shop.js

```diff
@@ -0,0 +1,207 @@
+    }
+
+    get bidButtonLocator(){
+        return ".shop-bid-button";
+    }
+
+    get listButton(){
+        return browser.driver.findElement(by.css(".shop-header-view span:last-child"));
+    }
+
+    get gridBidButton(){
+        return browser.driver.findElement(by.css(".overlay div"))
+    }
+    
+    filterNumber(number){
+        var value;
+        switch(number){
+            case 1:
+                value= browser.driver.findElement(by.css("#shop-filter-cat-single:nth-child(1)"));
+                return value;
+            case 2:
+                return browser.driver.findElement(by.css("#shop-filter-cat-single:nth-child(2)"));
+            case 3:
+                return browser.driver.findElement(by.css("#shop-filter-cat-single:nth-child(3)"));
+            case 4:
+                return browser.driver.findElement(by.css("#shop-filter-cat-single:nth-child(4)"));
+            case 5:
+                return browser.driver.findElement(by.css("#shop-filter-cat-single:nth-child(5)"));
+            case 6:
+                return browser.driver.findElement(by.css("#shop-filter-cat-single:nth-child(6)"));
+            case 7:
+                return browser.driver.findElement(by.css("#shop-filter-cat-single:nth-child(7)"));
+            case 8:
+                return browser.driver.findElement(by.css("#shop-filter-cat-single:nth-child(8)"));
+            case 9:
+                return browser.driver.findElement(by.css("#shop-filter-cat-single:nth-child(9)"));
+            case 10:
+                return browser.driver.findElement(by.css("#shop-filter-cat-single:nth-child(10)"));
+            
+            default:  return browser.driver.findElement(by.css("#shop-filter-cat-single:nth-child(2)"));
+            }
+    }
+
+    get leftSliderButton(){
+        return browser.driver.findElement(by.className("rc-slider-handle-1"));
+    }
+
+    get rightSliderButton(){
+        return browser.driver.findElement(by.className("rc-slider-handle-2"));
+    }
+
+    get jacketShopFilter(){
+        return browser.driver.findElement(by.css("#shop-filter-sub-list-inside :last-child"));
+    }
+
+    get dressShopFilter(){
+        return browser.driver.findElement(by.css("#shop-filter-sub-list-inside :first-child"))
+    }
+
+    get labelPricesFromFilterByPrice(){
+        return browser.driver.findElement(by.css("#shop-filters-price-prices label"));
+    }
+
+    get selectedFilterPopUp(){
+        return browser.driver.findElement(by.className("active-name"))
+    }
+
+    
+
+    // ACTIONS
+
+    getTextFromSelectedFilterPopUp(){
+        console.log("This method gets text from selected filter pop up");
+        return this.selectedFilterPopUp.getText();
+    }
+    validateElements(){
+        return this.getTextFromSelectedFilterPopUp()
+            .then((filterText) => this.validateTextFromSelectedFilterPopUp(filterText))
+    }
+    validateTextFromSelectedFilterPopUp(filterText){
```

Comment:
This method should receive one more parametar which will tell to whih data to comapare. For example:
 validateTextFromSelectedFilterPopUp(filterText, option){
        console.log("This method validates text from selected filter pop up");
        if(option) {
            return expect(filterText).toBe(data.titleJacket)
        }
            return expect(filterText).toBe(data.colorToSearch)
        }   
    }

Reply:
When you have IF ELSE , and inside return , then there is no need for else.
You can replace logic 
if(a) {
return something1: }
else { 
return something2:  }

with
if(a) {
return something1; }
return something2;


This repeats in your code , review once again.

Reply:
I changed logic for IF ELSE. 
And for the first comment, example you wrote is the best solution for when only two filters are considered (in this case, search by jacket, and search by color), but there are more filters in shop (shoes, dresses etc), so I thought that it's good thing to prepare method for maybe future development of the page where even more filters (like jewelry) will be included, so that I can simply pass more parameters (filters) to the methods, but if we're looking only at the current state, than I will refactor the method in a way that you wrote.